src/search/interval.h

```diff
@@ -128,11 +128,32 @@ struct IntervalSet {
   }
 
   friend IntervalSet operator&(const IntervalSet &l, const IntervalSet &r) {
-    if (l.IsEmpty() || r.IsEmpty()) {
-      return IntervalSet();
+    IntervalSet result;
+
+    if (l.intervals.empty() || r.intervals.empty()) {
+      return result;
+    }
+
+    auto it_l = l.intervals.begin();
+    auto it_r = r.intervals.begin();
+
+    while (it_l != l.intervals.end() && it_r != r.intervals.end()) {
+      // Find overlap between current intervals
```

Comment:
@PragmaTwice quick question, does our interval prevent from generating qNaN?

Reply:
We assume that there's no NaN in the input.